posthog/queries/sessions/session_recording.py

```diff
@@ -94,20 +103,32 @@ def filter_sessions_by_recordings(
     max_ts = max(it["end_time"] for it in sessions_results)
 
     session_recordings = query(team, min_ts, max_ts, filter)
+    viewed_session_recordings = set(
+        SessionRecordingViewed.objects.filter(team=team, user_id=filter.user_id).values_list("session_id", flat=True)
+    )
 
     for session in sessions_results:
-        session["session_recording_ids"] = [
-            recording["session_id"] for recording in session_recordings if matches(session, recording)
-        ]
+        session["session_recordings"] = list(
+            collect_matching_recordings(session, session_recordings, filter, viewed_session_recordings)
+        )
 
     if filter.limit_by_recordings:
-        sessions_results = [session for session in sessions_results if len(session["session_recording_ids"]) > 0]
+        sessions_results = [session for session in sessions_results if len(session["session_recordings"]) > 0]
```

Comment:
I think we have two separate for loops here iterating over the same content, maybe worth optimizing?

Reply:
This wouldn't really speed things up as and would be slightly unclearer.